docs/src/ref_impl/creating_appvm.md

```diff
@@ -0,0 +1,90 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Creating an Application VM
+
+## What is AppVM?
+
+AppVM is a virtual machine, which isolates an application from a host OS. An application seems like it is running inside an ordinary OS, yet it runs in a completely different system. An application inside renders, shows and controls from a host OS via GUI VM, network access is provided by NetVM.
+
+In the perfect scenario, there is a separate AppVM for every application. But it is possible to run multiple applications inside one AppVM.
+
+
+## How to add a new AppVM
+
+### 1. AppVM description
+
+Add the VM description in the target configuration.
+[lenovo-x1.nix](../../../targets/lenovo-x1.nix) already has AppVMs inside for Chromium, Gala, and Zathura applications.
+
+#### Example of the current AppVMs
+
+```
+vms = with pkgs; [
+  {
+    name = "chromium";
+    packages = [chromium];
+    ipAddress = "192.168.101.5/24";
+    macAddress = "02:00:00:03:03:05";
+    ramMb = 3072;
+    cores = 4;
+  }
+  {
+    name = "gala";
+    packages = [(pkgs.callPackage ../user-apps/gala {})];
+    ipAddress = "192.168.101.6/24";
+    macAddress = "02:00:00:03:03:06";
+    ramMb = 1536;
+    cores = 2;
+  }
+  {
+    name = "zathura";
+    packages = [zathura];
+    ipAddress = "192.168.101.7/24";
+    macAddress = "02:00:00:03:03:07";
+    ramMb = 512;
+    cores = 1;
+  }
+];
+```
+
+Each VM has the following properties:
+
+
+| **Property** | **Type**                  | **Unique** | **Description**                                                                                               | **Example**         |
+| -------------- | --------------------------- | ------------ | --------------------------------------------------------------------------------------------------------------- | --------------------- |
+| name         | str                       | yes        | This name will be shown in microvm list, a VM will have the same hostname                                     | “chromium”        |
```

Comment:
Should be (automatically) postfix the `name` with `-vm`? That would make it clearer that the application name is a virtual machine of its' own.

Reply:
It's actually prefixed here - https://github.com/tiiuae/ghaf/pull/243/files#diff-b7c4d8b1828ec4e55c3b85b348f4c180560550100c49b79a826d4cacfd1afaedR14
Maybe mention in the `Description` field of `name` that `This name is prefixed with vm- and will be shown in microvm list. The prefixed name - e.g. `vm-chromium` will be also the VM hostname.

Reply:
Added